app/Article.php

```diff
@@ -141,4 +143,19 @@ public function getReadingTimeAttribute()
     {
         return getEstimatedReadingTime($this->summary . ' ' . $this->content, \App::getLocale());
     }
+
+    public function getParsedContent()
+    {
+        $content = Str::of($this->content)->replaceMatches("/\[x-article_teaser id=&#39;.*&#39;\/\]/", function ($match) {
+            if (empty($match[0])) return '';
+            $id = Str::of($match[0])->between("id=&#39;","&#39;");
+            $article = Article::find($id);
```

Comment:
just put id into parenthesis in the regexp above and use

```suggestion
            $article = Article::find($match[1]);
```

Reply:
It should also only filter published articles (please add a scope)